pkg/apis/pipeline/v1alpha1/pipelinerun_types_test.go

```diff
@@ -281,3 +281,106 @@ func TestPipelineRunGetServiceAccountName(t *testing.T) {
 		}
 	}
 }
+
+func TestPipelineRunGetPodSpecSABackcompatibility(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		pr      *v1alpha1.PipelineRun
+		saNames map[string]string
+	}{
+		{
+			name: "test backward compatibility",
+			pr: &v1alpha1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{Name: "pr"},
+				Spec: v1alpha1.PipelineRunSpec{
+					PipelineRef:        &v1alpha1.PipelineRef{Name: "prs"},
+					ServiceAccountName: "defaultSA",
+					ServiceAccountNames: []v1alpha1.PipelineRunSpecServiceAccountName{{
+						TaskName: "taskName", ServiceAccountName: "taskSA",
+					}},
+					TaskRunSpecs: []v1alpha1.PipelineTaskRunSpec{{
+						PipelineTaskName:       "taskName",
+						TaskServiceAccountName: "newTaskSA",
+					}},
+				},
+			},
+			saNames: map[string]string{
+				"unknown":  "defaultSA",
+				"taskName": "newTaskSA",
+			},
+		},
+		{
+			name: "mixed default SA backward compatibility",
+			pr: &v1alpha1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{Name: "pr"},
+				Spec: v1alpha1.PipelineRunSpec{
+					PipelineRef:        &v1alpha1.PipelineRef{Name: "prs"},
+					ServiceAccountName: "defaultSA",
+					TaskRunSpecs: []v1alpha1.PipelineTaskRunSpec{{
+						PipelineTaskName:       "taskNameOne",
+						TaskServiceAccountName: "TaskSAOne",
+					}, {
+						PipelineTaskName:       "taskNameTwo",
+						TaskServiceAccountName: "newTaskTwo",
+					}},
+				},
+			},
+			saNames: map[string]string{
+				"unknown":     "defaultSA",
+				"taskNameOne": "TaskSAOne",
+				"taskNameTwo": "newTaskTwo",
+			},
+		},
+	} {
+		for taskName, expected := range tt.saNames {
```

Comment:
We should wrap each sub-test  inside a `t.Run()` call like https://github.com/tektoncd/pipeline/blob/ed0e1a0b37e6ff61bf32ce259d826e6d5dd820d7/pkg/apis/pipeline/v1alpha1/pipelinerun_types_test.go#L187

Makes it a lot easier to see what was run/what failed